nixos/modules/virtualisation/oci-containers.nix

```diff
@@ -372,16 +413,20 @@ let
             ${container.imageStream} | ${cfg.backend} load
           ''}
           ${optionalString (cfg.backend == "podman") ''
-            rm -f /run/podman-${escapedName}.ctr-id
+            rm -f /run/${escapedName}/ctr-id
           ''}
         '';
       };
+
+      effectiveUser = container.podman.user or "root";
+      dependOnLingerService =
+        cfg.backend == "podman" && effectiveUser != "root" && config.users.users.${effectiveUser}.linger;
     in
     {
       wantedBy = [ ] ++ optional (container.autoStart) "multi-user.target";
-      wants = lib.optional (
-        container.imageFile == null && container.imageStream == null
-      ) "network-online.target";
+      wants =
+        lib.optional (container.imageFile == null && container.imageStream == null) "network-online.target"
+        ++ lib.optional dependOnLingerService "linger-users.service";
```

Comment:
Still kinda amazing, how this is the result of an RFC that has

> We want a unified Nix code style that's consistent with itself, easily readable, accessible and __results in readable diffs.__

(emphasis mine) as explicit goal.